### DISCUSSION TITLE
Fill out pdf with form fields?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [pymupdf/PyMuPDF#3846] [Looking for help]
Hello - i have the attached pdf with form fields templ.pdf

Is it somehow possible to fillout the form-fields and save a new pdf with that?

[templ.pdf](https://github.com/user-attachments/files/16873197/templ.pdf)

**Comment by JorjMcKie** [comment]
Sure - please see the documentation [here](https://pymupdf.readthedocs.io/en/latest/widget.html#Widget). You can do `widget.field_value = something` and then call `widget.update()`.

**Comment by JorjMcKie** [comment]
The widgets of a page can be accessed via the iterator `page.widgets()`. You can also start with the first widget on the page `page.first_widget`. Then access the next widget via its `.next` property. The end of this chain is reached if `widget.next is None`.

**Comment by Rapid1898-code** [comment]
Thx again for the fast answer!
